package-lock.json

```diff
@@ -3830,8 +3830,7 @@
         },
         "ansi-regex": {
           "version": "2.1.1",
-          "bundled": true,
-          "optional": true
+          "bundled": true
```

Comment:
@jorgeorpinel what version of npm do you have? it looks like the difference here is only in some defaults. It can happen if a different version of npm is used.

Reply:
6.4.1

Reply:
How do you manage your Node version? Seems like the official way to upgrade is to download the installer every time... Have you tried [n](https://www.npmjs.com/package/n)?

Reply:
@shcheklein Interesting. What I can get from the changes is that `package-lock.json` on my machine marked dependencies for `fsevents` as optional, but this one marked them as required. Question is why it marked them as optional in the first place.

`fsevents` is a library used on MacOS to detect file changes then you for example need to rebuild your bundle is some files in FS changes. It is used only on Mac, requires creating C bindings with `node-gyp` and if bindings are not created it can usually still work without them using manual polling. So yes, if @jorgeorpinel is on Mac and tried to run `npm run dev` it can potentially break it.

I don't have a good explanation to why have they been marked optional in the first place through. According to `git blame` it was added in my commit f9c1415e5aac0b262279570dd4c8bffa121f7aa3, but this commit only added `lodash.debounce` so theoretically it shouldn't affect `fsevents` dependencies at all. There is also no info about package-lock.json bugs in `npm` changelog.

So yes, this version without `optional` flag is correct version and we should used it, there is no need to change npm vesrion.

About what to do with it to prevent bugs like hat in the future my only idea in to move to `yarn` from `npm`. We are using yarn and `yarn.lock` files on our projects by default for the last two years or so and there was no such problems with them.

About updating node and npm, and using different node version: On Mac and Linux using package managers like https://brew.sh/ should solve problem with updates. For Windows there is https://chocolatey.org/ , but I never used it myself, so I can't comment on it. And to have multiple versions at the same time we usually use https://github.com/nvm-sh/nvm and not `n`.

Reply:
Found relevant issue in NPM by the way: https://github.com/npm/npm/issues/17722

Reply:
Created ticket #595

Reply:
Should be resolved by migrating to yarn. Please, merge and resolve the conflicts.